wrTransport.c

```diff
@@ -0,0 +1,242 @@
+#include "wrTransport.h"
+
+#include <stdlib.h>
+#include "wrMath.h"
+
+/**
+  transport_init():
+    [TODO: description]
+**/
+uint8_t transport_init( transport_t* self, uint16_t b_size )
+{
+    uint8_t error = 0;
+
+    self->active        = 0;
+    self->tape_islocked = 0;
+
+    // default speed values
+    self->speeds = (std_speeds_t) {
+         .max_speed = 2.0
+       , .accel_standard = 0.001
+       , .accel_quick = 0.05
+       , .accel_seek = 0.001
+       , .accel_nudge = 0.005
+       , .nudge_release = 0.002
+     };
+
+    lp1_init(      &(self->speed_slew) );
+    lp1_set_coeff( &(self->speed_slew), (self->speeds).accel_standard );
+
+    self->b_size = b_size;
+    uint16_t speed_len = (b_size + 1);
+    self->speed_v = malloc(sizeof(float) * speed_len );
+
+    if( self->speed_v == NULL ){ error = 1; }
+    for( uint16_t i=0; i<speed_len; i++ ){ self->speed_v[i] = 0.0; }
+    self->speed_play   = 1.0;
+    self->speed_stop   = 0.0;
+
+    lp1_init(      &(self->speed_manual) );
+    lp1_set_coeff( &(self->speed_manual), (self->speeds).accel_seek );
+    self->nudge        = 0.0;
+    self->nudge_accum  = 0.0;
+
+    return error;
+}
+
+/**
+  transport_deinit():
+    [TODO: description]
+**/
+void transport_deinit( transport_t* self )
+{
+  free(self->speed_v);
+  free(self);
```

Comment:
`deinit` shouldn't free the `transport_t` itself, because `init` didn't malloc it.

do you have ideas on whether `init` *should* malloc the transport_t and return a pointer to that struct?